diff --git a/transformer_engine/pytorch/setup.py b/transformer_engine/pytorch/setup.py
index be6a1d75..40217c52 100644
--- a/transformer_engine/pytorch/setup.py
+++ b/transformer_engine/pytorch/setup.py
@@ -14,6 +14,7 @@ from pathlib import Path
 import setuptools
 
 try:
+    import torch
     from torch.utils.cpp_extension import BuildExtension
 except ImportError as e:
     raise RuntimeError("This package needs Torch to build.") from e
@@ -37,6 +38,8 @@ from build_tools.pytorch import setup_pytorch_extension
 os.environ["NVTE_PROJECT_BUILDING"] = "1"
 CMakeBuildExtension = get_build_ext(BuildExtension, True)
 
+TORCH_MAJOR = int(torch.__version__.split(".")[0])
+TORCH_MINOR = int(torch.__version__.split(".")[1])
 
 if __name__ == "__main__":
     # Extensions
@@ -56,7 +59,7 @@ if __name__ == "__main__":
         ext_modules=ext_modules,
         cmdclass={"build_ext": CMakeBuildExtension},
         setup_requires=[
-            "torch>=2.1",
+            f"torch=={TORCH_MAJOR}.{TORCH_MINOR}.*",
             "nvidia-cuda-runtime-cu12",
             "nvidia-cublas-cu12",
             "nvidia-cudnn-cu12",
@@ -65,7 +68,7 @@ if __name__ == "__main__":
             "nvidia-nvtx-cu12",
             "nvidia-cuda-nvrtc-cu12",
         ],
-        install_requires=["torch>=2.1"],
+        install_requires=[f"torch=={TORCH_MAJOR}.{TORCH_MINOR}.*"],
         tests_require=["numpy", "torchvision"],
     )
     if any(x in sys.argv for x in (".", "sdist", "bdist_wheel")):
