diff --git a/transformer_engine/pytorch/setup.py b/transformer_engine/pytorch/setup.py
index be6a1d75..723783b5 100644
--- a/transformer_engine/pytorch/setup.py
+++ b/transformer_engine/pytorch/setup.py
@@ -19,6 +19,9 @@ except ImportError as e:
     raise RuntimeError("This package needs Torch to build.") from e
 
 
+TORCH_MAJOR = int(torch.__version__.split(".")[0])
+TORCH_MINOR = int(torch.__version__.split(".")[1])
+
 current_file_path = Path(__file__).parent.resolve()
 build_tools_dir = current_file_path.parent.parent / "build_tools"
 if bool(int(os.getenv("NVTE_RELEASE_BUILD", "0"))) or os.path.isdir(build_tools_dir):
@@ -56,7 +59,7 @@ if __name__ == "__main__":
         ext_modules=ext_modules,
         cmdclass={"build_ext": CMakeBuildExtension},
         setup_requires=[
-            "torch>=2.1",
+            f"torch=={TORCH_MAJOR}.{TORCH_MINOR}.*",
             "nvidia-cuda-runtime-cu12",
             "nvidia-cublas-cu12",
             "nvidia-cudnn-cu12",
@@ -65,7 +68,7 @@ if __name__ == "__main__":
             "nvidia-nvtx-cu12",
             "nvidia-cuda-nvrtc-cu12",
         ],
-        install_requires=["torch>=2.1"],
+        install_requires=[f"torch=={TORCH_MAJOR}.{TORCH_MINOR}.*"],
         tests_require=["numpy", "torchvision"],
     )
     if any(x in sys.argv for x in (".", "sdist", "bdist_wheel")):
